BigInt.cpp

```diff
@@ -0,0 +1,163 @@
+    }
+    return os;
+}
+std::istream & operator>>(std::istream &is , BigInt &bigInt){
+    std::string number;
+    is >> number;
+    int length = number.length();
+    int partCount = ceil(length / 9.0);
+    for (int i = 0; i < partCount; ++i) {
+        int endIndex = length - i * 9;
+        int startIndex = std::max(endIndex - 9, 0);
+        bigInt.number[i] = std::stoi(number.substr(startIndex, endIndex - startIndex));
+    }
+    
+    return is;
+}
+bool BigInt::operator==(const BigInt &bigInt){
+    for(int i = 0 ; i < 5 ; i++){
+        if(this->number[i] != bigInt.number[i]){
+            return false;
+        }
+    }
+    return true;
+}
+bool BigInt::operator!=(const BigInt &bigInt){
+        for(int i = 0 ; i < 5 ; i++){
+        if(this->number[i] != bigInt.number[i]){
+            return true;
+        }
+    }
+    return false;
+}
+bool BigInt::operator<(const BigInt &bigInt){
+    for (int i = 4; i >= 0; i--) {
+        if (this->number[i] < bigInt.number[i]) {
+            return true; 
+        } else if (this->number[i] > bigInt.number[i]) {
+            return false; 
+        }
+    }
+    return false; 
+}
+bool BigInt::operator<=(const BigInt &bigInt){
+    for (int i = 4; i >= 0; i--) {
+        if(this->number[i] == bigInt.number[i]){
+            continue;
+        } else if (this->number[i] < bigInt.number[i]) {
+            return true; 
+        } else {
+            return false; 
+        }
+    }
+    return true;
+}
+bool BigInt::operator>(const BigInt &bigInt){
+    for (int i = 4; i >= 0; i--) {
+        if (this->number[i] > bigInt.number[i]) {
+            return true; 
+        } else if (this->number[i] < bigInt.number[i]) {
+            return false; 
+        }
+    }
+    return true; 
+}
+bool BigInt::operator>=(const BigInt &bigInt){
+        for (int i = 4; i >= 0; i--) {
+        if(this->number[i] == bigInt.number[i]){
+            continue;
+        } else if (this->number[i] > bigInt.number[i]) {
+            return true; 
+        } else {
+            return false; 
+        }
+    }
+    return true;
+}
+BigInt BigInt::operator+(const BigInt &bigInt){
+    BigInt returnBigInt;
+    for(int i = 0 ; i < 5 ; i++){
+        returnBigInt.number[i] += this->number[i] + bigInt.number[i];
+        if(returnBigInt.number[i] >= pow(10 , 9)){
+            if(i == 4){
+                throw "out of range";
+            }
+            returnBigInt.number[i+1]++;
+            returnBigInt.number[i] -= pow(10,9);
+        } 
+    }
+    return returnBigInt;
+}
+BigInt BigInt::operator-(const BigInt &bigInt){
+    BigInt returnBigInt;
+    for(int i = 0 ; i < 5 ; i++){
+        returnBigInt.number[i] += this->number[i] - bigInt.number[i];
+        if(returnBigInt.number[i] < 0 ){
+            if(i == 4){
+                throw "out of range";
+            }
+            returnBigInt.number[i+1]--;
+            returnBigInt.number[i] += pow(10,9);
+        } 
+    }
+    return returnBigInt;
+}

```

Comment:
Same here, the logic is the same, the only difference is `* (-1)` in some places if it's the `-` operator. Try to write a function and use it in both operators.